Add support for Thunderbird addons gallery

Jackymancs4 opened the pull request:
Hello @Rob--W 
Following your work on https://github.com/Rob--W/crxviewer/commit/ecbe019a990bcce7c7f54093befecd0a413fe16a for https://github.com/Rob--W/crxviewer/issues/72, this PR add pageAction support for the [Thunderbird addons gallery](https://addons.thunderbird.net).

Using the contextual menu already worked, but this seemed a +1 on usability to me.
Since https://github.com/thundernest/addons-server is a fork of the one used by Mozilla, I upgraded the amo regexes instead of adding new ones.

Also, it is present quite a bit of name hardcoding. I hope it's fine, considering I tryied to match what was already there.

Tested in Firefox 64.0 and Chrome 71.0.3578.98

Rob--W reviewed (changes requested):
Hey, thanks for this PR! I have left some review comments below.

And in this PR, everything is converted to Thunderbird where possible, except for the search function at: https://github.com/Rob--W/crxviewer/blob/0c8a5723863a7e497f6115bf2a59f69c53d49297/src/crxviewer.js#L1627-L1631

By default `get_amo_domain` returns `addons.mozilla.org`, so if there is no URL given at all, the AMO search form will search on AMO. Leaving it at that is OK; if you want to have different behavior feel free to propose reasonable patches.

Jackymancs4 commented:
I had to revert and force push to properly clean the whitespace changes.

I refactored the exact way you suggested, doing basic testing by hand afterwards. If needed, I can test more extensively tomorrow. Still, It works fine for every use case of mine.

> And in this PR, everything is converted to Thunderbird where possible, except for the search function at: /src/crxviewer.js@0c8a572#L1627-L1631

> Leaving it at that is OK

Agree. Right now defaulting to addons.mozilla.org is still a sane strategy, and speaking for myself I don't use that feauture.

Rob--W commented on src/cws_pattern.js:
```diff
@@ -162,7 +165,7 @@ function get_webstore_url(url) {
     var amo = get_amo_slug(url);
     if (amo) {
-        return 'https://' + get_amo_domain(url) + '/firefox/addon/' + amo;
+        return 'https://' + get_amo_domain(url) + '/'+(get_amo_domain(url)=="addons.thunderbird.net" ? "thunderbird" : "firefox")+'/addon/' + amo;
```
Use `get_amo_base_url(get_amo_domain(url))` here.

Rob--W commented on src/cws_pattern.js:
```diff
@@ -70,7 +73,7 @@ function get_xpi_url(amoDomain, addonSlug) {
     }
 
-    var url = 'https://' + amoDomain + '/firefox/downloads/latest/';
+    var url = get_amo_base_url(amoDomain)+'/downloads/latest/';
```
Add spaces around `+`.

Rob--W commented on src/cws_pattern.js:
```diff
@@ -96,7 +99,7 @@ function get_crx_url(extensionID_or_url) {
     match = amo_file_version_pattern.exec(extensionID_or_url);
     if (match) {
-        return 'https://' + match[1] + '/firefox/downloads/file/' + match[2] + (match[3] || '/addon.xpi');
+        return get_amo_base_url(match[1])+'/downloads/file/' + match[2] + (match[3] || '/addon.xpi');
```
Add spaces around `+`.

Rob--W commented on src/cws_pattern.js:
```diff
@@ -182,6 +185,10 @@ function get_zip_name(url, /*optional*/filename) {
 }
 
+function get_amo_base_url(amoDomain) {
+    return 'https://' + amoDomain + '/' + (amoDomain=="addons.thunderbird.net" ? "thunderbird" : "firefox")
```
Please be consistent in use of `'` and `"`, and whitespace around operators. Change this to:

```javascript
if (amoDomain === 'addons.thunderbird.net') {
    return 'https://addons.thunderbird.net/thunderbird';
}
return 'https://' + amoDomain + '/firefox';
```

Rob--W reviewed (changes requested):
Thanks for the follow-up. Could you make the following changes and squash the commits + choose a good commit title (your PR title is good)?

Jackymancs4 committed:
Add support for thunderbird addons gallery

Add trail commas

Fixed regex amo_file_version_pattern and refactored amo url crafting

Enforce consistency, minor get_amo_base_url refactor

benjaoming commented:
@Rob--W any possibility to merge and release this?
